rllib/algorithms/algorithm_config.py

```diff
@@ -3659,6 +3663,10 @@ def reporting(
             log_gradients: Log gradients to results. If this is `True` the global norm
                 of the gradients dictionariy for each optimizer is logged to results.
                 The default is `False`.
+            custom_stats_cls_lookup: A dictionary mapping stat names to their corresponding Stats classes.
+                This allows you to use your own Stats classes for logging metrics.
+                The keys of the dictionary are the stat names, and the values are the corresponding Stats classes.
+                The Stats classes must be subclasses of rllib.utils.metrics.StatsBase.
```

Comment:
Maybe add what happens when using stat names that already exist like "mean" and what happens if names are not used at all in RLlib - very custom and maybe used in a custom callback by the user.